### PR TITLE
feat: add network fetching from runner

### PR DESCRIPTION
**Pull request by ndaWefox**

### CODE REVIEW

File: rustainers/src/runner/inner.rs [hunk 1]
```diff
@@ -40,6 +41,14 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
         Ok(result)
     }
 
+    #[tracing::instrument(level = "debug", skip(self), fields(runner = %self))]
+    async fn list_networks(&self) -> Result<Vec<RunnerNetwork>, ContainerError> {
+        let mut cmd = self.command();
+        cmd.push_args(["network", "ls", "--no-trunc", "--format={{json .}}"]);
+        let result = cmd.json_stream::<RunnerNetwork>().await?;
+        Ok(result)
+    }
```
Comment (davidB) [inline rustainers/src/runner/inner.rs:50]: I guess that you don't want to list all networks but the one used by the container. So may you should use `inspect` (and enhance existing related function)

File: rustainers/src/runner/inner.rs [hunk 2]
```diff
@@ -40,6 +41,14 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
         Ok(result)
     }
 
+    #[tracing::instrument(level = "debug", skip(self), fields(runner = %self))]
```
Comment (IgorLaborieWefox) [inline rustainers/src/runner/inner.rs:44]: 🤌 **nitpick**: move this function below ``create_network`
I prefer have command grouped by topics

File: rustainers/src/runner/runner_network.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    fn should_serde_network() {
+        let json_stream = include_str!("../../tests/assets/docker_networks.jsonl");
+        let stream = serde_json::Deserializer::from_str(json_stream).into_iter::<RunnerNetwork>();
+        let networks = stream.collect::<Result<Vec<_>, _>>();
+        let_assert!(Ok(data) = networks);
+        insta::assert_debug_snapshot!(data);
+    }
+
+    #[test]
+    fn should_filter_network() {
```
Comment (IgorLaborieWefox) [inline rustainers/src/runner/runner_network.rs:59]: 🔨 **todo**: Also needs to be tested with podman.

=> I will give you a sample file

**Review by IgorLaborieWefox — CHANGES_REQUESTED** [review]
Just require a test with podman.

File: rustainers/src/runner/inner.rs (hunk 1 above)
Comment (ndaWefox) [reply rustainers/src/runner/inner.rs:50]: Actually, I want to list all the network for a given runner and not just for a given container.
The intent is to have the ability to run a container inside another container created in the context of a given network. That allows us to solve the docker in docker for github ci

File: rustainers/src/runner/runner_network.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
+use serde::{Deserialize, Serialize};
+
+/// The network driver
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Driver {
```
Comment (IgorLaborieWefox) [inline rustainers/src/runner/runner_network.rs:6]: 🔨 **todo**: There is a similar type `crate::container::network::Network`
Null -> None
Ipvlan, & macvlan -> Custom ?
Overlay ? maybe custom, maybe this is a container ?

File: rustainers/src/runner/runner_network.rs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    Macvlan,
+    /// Null
+    Null,
+}
+
+/// A runner network
+///
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct RunnerNetwork {
```
Comment (IgorLaborieWefox) [inline rustainers/src/runner/runner_network.rs:25]: 🔨 **todo**: This type could be moved into the `crate::container::network` module.
Also move the related tests to remove this file

**Review by IgorLaborieWefox — CHANGES_REQUESTED** [review]

File: rustainers/src/runner/runner_network.rs (hunk 4 above)
Comment (IgorLaborieWefox) [inline rustainers/src/runner/runner_network.rs:6]: Add #[non_exhaustive]

**Comment by ndaWefox** [comment]
Closing this as [PR](https://github.com/wefoxplatform/rustainers/pull/24) is tackling this

**Pull request closed without merging**